hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerRatisServer.java

```diff
@@ -330,14 +330,45 @@ public void addOMToRatisRing(OMNodeDetails newOMNode) throws IOException {
     }
   }
 
+  /**
+   * Remove decommissioned OM node from Ratis ring
+   */
+  public void removeOMFromRatisRing(OMNodeDetails removeOMNode)
+      throws IOException {
+    Preconditions.checkNotNull(removeOMNode);
+
+    String removeNodeId = removeOMNode.getNodeId();
```

Comment:
Do we want to allow only operation at a time on this operation? And throw exception to other invokes on this API from client Like Already remove is ongoing something like this.

Coming from the perspective of one cli remove om1, and another cli remove om2. If we know inprogress thing, we can ask for 2nd invocation retry once after first is completed. Do we need any such semantics here?

Reply:
And also one question how remove works when leader OM needs to be removed? 
And when remove in setConf we update peerlist, and add/remove grpcAppender Threads.

Reply:
I think after reading test understood, we can stop and then decomission. So, we don't need to worry about that if leader is the one being removed i think. Is my understanding correct?

Reply:
With the current patch, we cannot remove leader OM. So yes, OM can be stopped before decommissioning. But if there are only 2 OMs in the ring, and we stop the follower to decommission, then the Leader will not be able to process the SetConf request as it won't get the quorum.
So, the more general rule would be that decommissioning OM should not be the leader.

Reply:
if we want to remove a leader from the ratis ring, can we first transfer the leadership to a follower and then remove the old leader(now it is follower)

Reply:
@JacksonYao287, yes that should be possible. As long as a node is not currently the leader, it can be decommissioned.